vitrina/datasets/views.py

```diff
@@ -478,3 +484,70 @@ def get_success_url(self):
         return reverse('dataset-members', kwargs={
             'pk': self.kwargs.get('dataset_id'),
         })
+
+
+class DatasetStatsView(DatasetListView):
+    facet_fields = ['filter_status', 'organization', 'category', 'frequency', 'tags', 'formats']
+    template_name = 'vitrina/datasets/statistics_graph.html'
+    paginate_by = 0
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        data = []
+        dataset_ids = []
+        status_translations = {}
+        status_graph_styles = {'INVENTORED': {'borderColor': 'black',
+                                              'backgroundColor': 'rgba(255, 179, 186, 0.9)',
+                                              'order': 3},
+                               'OPENED': {'borderColor': 'black',
+                                          'backgroundColor': 'rgba(255, 223, 186, 0.9)',
+                                          'order': 2},
+                               'STRUCTURED': {'borderColor': 'black',
+                                              'backgroundColor': 'rgba(186,225,255, 0.9)',
+                                              'order': 1}}
+
+        for dataset in context.get('object_list'):
+            dataset_ids.append(dataset.pk)
```

Comment:
Please, use a single query, to get the stats, constructing queries from data is doomed to fail.

My quick guess is that you probably can do this, by constructing a subquery from comments and then doing an inner join on man query, to filter out, datasets that are not in main query. Something like that.